domain/tests/src/Unit/Access/DomainAccessCheckTest.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Drupal\Tests\domain\Unit\Access;
+
+//use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\domain\Access\DomainAccessCheck;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Tests the domain record actions.
+ *
+ * @group domain
+ */
+class DomainAccessCheckTest extends UnitTestCase {
+
+  /** @var \Drupal\domain\Access\DomainAccessCheck */
+  private $object;
+
+  /** @var \PHPUnit_Framework_MockObject_MockObject */
+  private $mockNegotiator;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $negotiator = $this->getMockBuilder('\Drupal\domain\DomainNegotiator')
+      ->disableOriginalConstructor()
+      ->getMock();
+    $this->mockNegotiator = $negotiator;
+    $this->object = new DomainAccessCheck($negotiator);
+  }
+
+  /**
+   * @param string $path
+   *
+   * @dataProvider providerCheckPathTrue
+   */
+  public function testAppliesTrue($path) {
+    $route = $this->getMockBuilder('\Symfony\Component\Routing\Route')
+      ->disableOriginalConstructor()
+      ->getMock();
+    $route->expects(static::once())->method('getPath')
+      ->willReturn($path);
+    static::assertTrue($this->object->applies($route));
+  }
+
+  /**
+   * @param string $path
+   *
+   * @dataProvider providerCheckPathFalse
+   */
+  public function testAppliesFalse($path) {
+    $route = $this->getMockBuilder('\Symfony\Component\Routing\Route')
+      ->disableOriginalConstructor()
+      ->getMock();
+    $route->expects(static::once())->method('getPath')
+      ->willReturn($path);
+    static::assertFalse($this->object->applies($route));
+  }
+
+  /**
+   * @param string $path
+   *
+   * @dataProvider providerCheckPathTrue
+   */
+  public function testCheckPathTrue($path) {
+    static::assertTrue($this->object->checkPath($path));
+  }
+
+  /**
+   * @param string $path
+   *
+   * @dataProvider providerCheckPathFalse
+   */
+  public function testCheckPathFalse($path) {
+    static::assertFalse($this->object->checkPath($path));
+  }
+
+  /**
+   * @todo Find a way to mock AccessResult::allowedIfHasPermissions to complete test
+   *
+   * If AccessResult::allowedIfHasPermissions cannot be done by mock, then
+   * it will need to be done as a web test case. Maybe an example in comment\Unit\Entity\CommentLockTest
+   */
+  public function testAccess() {
```

Comment:
Going along with your comment below, I tend to agree that there is a point of vanishing returns. I may look at whether the the non-trivial tests can be done as a KernelTest -- might be a good deal more portable. (But I will have to become familiar with kernel tests to do that). I have deleted all that commented code and currently don't plan to try implementing those user checks as a unit test. Depending on what I find about kernel tests, the tests here might just move to a kernel test.